lib/aac/utils.js

```diff
@@ -24,13 +24,35 @@ var ADTS_SAMPLING_FREQUENCIES = [
   7350
 ];
 
+var parseId3TagSize = function(header, byteIndex) {
+  var
+    returnSize = (header[byteIndex + 6] << 21) |
+                 (header[byteIndex + 7] << 14) |
+                 (header[byteIndex + 8] << 7) |
+                 (header[byteIndex + 9]),
+    flags = header[byteIndex + 5],
+    footerPresent = (flags & 16) >> 4;
+
+  if (footerPresent) {
+    return returnSize + 20;
+  }
+  return returnSize + 10;
+};
+
+// TODO: use vhs-utils
 var isLikelyAacData = function(data) {
-  if ((data[0] === 'I'.charCodeAt(0)) &&
+  var offset = 0;
+
+  if (data.length > 10 && (data[0] === 'I'.charCodeAt(0)) &&
       (data[1] === 'D'.charCodeAt(0)) &&
       (data[2] === '3'.charCodeAt(0))) {
-    return true;
+    offset = parseId3TagSize(data, 0);
   }
-  return false;
+
+  return data.length >= offset + 2 &&
+    (data[offset] & 0xFF) === 0xFF &&
+    (data[offset + 1] & 0xE0) === 0xE0 &&
+    (data[offset + 1] & 0x16) === 0x10;
```

Comment:
May be worth a comment above that mentions the syncword that's being checked for and that this comparison verifies the 2 layer bits as always 0.